src/utils/shorten-url.js

```diff
@@ -4,65 +4,66 @@
   let longUrl = urlToShorten;
   if (!longUrl.startsWith('https://profiler.firefox.com/')) {
-    const parsedUrl = url.parse(longUrl);
-    const parsedUrlOnCanonicalHost = Object.assign({}, parsedUrl, {
-      protocol: 'https:',
-      host: 'profiler.firefox.com',
-    });
-    longUrl = url.format(parsedUrlOnCanonicalHost);
+    const parsedUrl = new URL(longUrl);
+    parsedUrl.protocol = 'https';
+    parsedUrl.host = 'profiler.firefox.com';
+    parsedUrl.port = '';
+    longUrl = parsedUrl.toString();
   }
 
-  const bitlyQueryUrl =
-    'https://api-ssl.bitly.com/v3/shorten?' +
-    queryString.stringify({
-      longUrl,
-      domain: 'perfht.ml',
-      format: 'json',
-      access_token: accessToken,
-    });
+  const bitlyQueryUrl = 'https://api-ssl.bitly.com/v4/shorten';
+  const payload = {
+    long_url: longUrl,
+    domain: 'perfht.ml',
+  };
 
-  return fetch(bitlyQueryUrl)
-    .then(response => response.json())
-    .then(json => json.data.url);
-}
+  const response = await fetch(bitlyQueryUrl, {
+    method: 'POST',
+    headers: {
+      Authorization: `Bearer ${accessToken}`,
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify(payload),
+  });
 
-export async function expandUrl(urlToExpand: string): Promise<string> {
-  const bitlyQueryUrl =
-    'https://api-ssl.bitly.com/v3/expand?' +
-    queryString.stringify({
-      shortUrl: urlToExpand,
-      format: 'json',
-      access_token: accessToken,
-    });
-  const response = await fetch(bitlyQueryUrl);
-  const json = await response.json();
-
-  if (!json.data) {
-    // In case of an error, json.data is null.
+  if (!response.ok) {
     throw new Error(
-      `An error happened while expanding the shortened url ${urlToExpand}: ${
-        json.status_txt
-      } (${json.status_code})`
+      `An error happened while shortening the long url ${longUrl}: ${
+        response.statusText
+      } (${response.status})`
     );
   }
 
-  const [data] = json.data.expand;
-  if (!response) {
+  const json = await response.json();
+  return json.link;
+}
+
+export async function expandUrl(urlToExpand: string): Promise<string> {
+  const bitlyQueryUrl = 'https://api-ssl.bitly.com/v4/expand';
+  const payload = {
+    bitlink_id: urlToExpand.replace(/^https:\/\//, ''),
+  };
+  const response = await fetch(bitlyQueryUrl, {
+    method: 'POST',
+    headers: {
+      Authorization: `Bearer ${accessToken}`,
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify(payload),
+  });
+
+  if (!response.ok) {
     throw new Error(
-      'There were no data in the otherwise well-formed answer from bit.ly.'
+      `An error happened while expanding the shortened url ${urlToExpand}: ${
+        response.statusText
+      } (${response.status})`
     );
   }
 
-  const longUrl = data.long_url;
-  if (!longUrl) {
-    throw new Error(`The short URL ${urlToExpand} couldn't be expanded.`);
-  }
```

Comment:
I removed this specific error case because this is now handled by a `404` HTTP response, which is already handled by the previous condition.